contrib/http_util/src/main/java/io/opencensus/contrib/http/AbstractHttpHandler.java

```diff
@@ -99,8 +111,37 @@ public final void handleMessageReceived(Span span, long messageId, long messageS
   public void handleEnd(Span span, @Nullable P response, @Nullable Throwable error) {
     checkNotNull(span, "span");
     int statusCode = extractor.getStatusCode(response);
-    span.putAttribute("http.status_code", AttributeValue.longAttributeValue(statusCode));
+    if (span.getOptions().contains(Options.RECORD_EVENTS)) {
+      span.putAttribute(HTTP_STATUS_CODE, AttributeValue.longAttributeValue(statusCode));
+    }
     span.setStatus(HttpTraceUtil.parseResponseStatus(statusCode, error));
     span.end();
   }
+
+  final String getSpanName(Q request, HttpExtractor<Q, P> extractor) {
+    // default span name
+    String path = extractor.getPath(request);
+    if (path == null) {
+      path = "/";
+    }
+    if (!path.startsWith("/")) {
+      path = "/" + path;
+    }
+    return path;
+  }
+
+  private static void putAttributeIfNotEmptyOrNull(Span span, String key, @Nullable String value) {
+    if (value != null && !value.isEmpty()) {
+      span.putAttribute(key, AttributeValue.stringAttributeValue(value));
+    }
+  }
+
+  final void addSpanRequestAttributes(Span span, Q request, HttpExtractor<Q, P> extractor) {
```

Comment:
if (span.getOptions().contains(Options.RECORD_EVENTS))  { ... } ? You will call this on the server as well.

Reply:
yes.